src/container/container.go

```diff
@@ -0,0 +1,272 @@
+/*
+Copyright © 2020 The PES Open Source Team pesos@pes.edu
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package container
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/client"
+	ui "github.com/gizak/termui/v3"
+	"github.com/pesos/grofer/src/general"
+)
+
+type PerContainerMetrics struct {
+	// Metrics common for Overall and per container
+	ID     string
+	Image  string
+	Name   string
+	Status string
+	State  string
+	Cpu    float64
+	Mem    float64
+	Net    netStat
+	Blk    blkStat
+	// Metrics specific to per container
+	Pid     string
+	NetInfo []netInfo
+	PerCPU  []string
+	PortMap []portMap
+	Mounts  []mountInfo
+	Procs   []procInfo
+}
+
+type netStat struct {
+	Rx float64
+	Tx float64
+}
+
+type blkStat struct {
+	Read  uint64
+	Write uint64
+}
+
+type netInfo struct {
+	Name    string
+	Driver  string
+	Ip      string
+	Ingress bool
+}
+
+type mountInfo struct {
+	Src  string
+	Dst  string
+	Mode string
+}
+
+type portMap struct {
+	Host      int
+	Container int
+	Protocol  string
+}
+
+type procInfo struct {
+	UID string
+	PID string
+	CMD string
+}
+
+// GetContainerMetrics provides per container metrics in the form of PerContainerMetrics Structs
+func GetContainerMetrics(cid string) (PerContainerMetrics, error) {
+
+	metrics := PerContainerMetrics{}
+
+	ctx := context.Background()
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		return metrics, err
+	}
+
+	// Get container using a filter
+	args := filters.NewArgs(
+		filters.KeyValuePair{
+			Key:   "id",
+			Value: cid,
+		},
+	)
+
+	containers, err := cli.ContainerList(ctx, types.ContainerListOptions{Filters: args})
+	if err != nil {
+		return metrics, err
+	}
+
+	if len(containers) > 1 {
+		return metrics, fmt.Errorf("multiple containers with same ID exist")
+	} else if len(containers) < 1 {
+		return metrics, general.ErrInvalidContainer
+	}
+
+	c := containers[0]
+
+	// Get PID
+	inspectData, err := cli.ContainerInspect(ctx, cid)
+	if err != nil {
+		return metrics, nil
+	}
+
+	// Get Container Stats
+	stats, _ := cli.ContainerStats(ctx, cid, false)
+	data := types.StatsJSON{}
+	err = json.NewDecoder(stats.Body).Decode(&data)
+	if err != nil {
+		return metrics, err
+	}
+
+	// Calculate CPU  and per CPU percent
```

Comment:
Can calculation of CPU% be in a different function? We could write tests for it that way.

Reply:
Done and tests added!